src/cdp/domains/dom.zig

```diff
@@ -266,10 +277,15 @@ fn resolveNode(cmd: anytype) !void {
     var js_context = page.main_context;
     if (params.executionContextId) |context_id| {
         if (js_context.v8_context.debugContextId() != context_id) {
-            var isolated_world = bc.isolated_world orelse return error.ContextNotFound;
-            js_context = &(isolated_world.executor.js_context orelse return error.ContextNotFound);
-
-            if (js_context.v8_context.debugContextId() != context_id) return error.ContextNotFound;
+            var found = false;
+            for (bc.isolated_worlds.items) |*isolated_world| {
+                js_context = &(isolated_world.executor.js_context orelse return error.ContextNotFound);
+                if (js_context.v8_context.debugContextId() == context_id) {
+                    found = true;
+                    break;
+                }
+            }
```

Comment:
You can drop the `found` variable and use `else` on the for loop.

```zig
for (....) {
} else {
    return error.ContextNotFound;
}
```

I think it's a python thing

Reply:
the `else` on a for executes only if the for loop naturally reaches its end. When a match is found, you'll `break` out and skip the `else`.

```zig
for (haystack) |value| {
   if (std.mem.eql(u8, value, needle)) {
      break;
   }
} else return error.NotFound;

// if you reach this point, you know `needle` was found in `haystack`
```